### DISCUSSION TITLE
How to Create Action Return IActionResult  in Endpoint

### DISCUSSION THREAD

**Discussion by eissa1005** [serenity-is/Serenity#5800] [Help] [answered]
When Create this occur error  in  like Service.ts Like ProductService.ts

**Comment by VictorTomaili** [comment, marked as answer]
Currently, `Endpoints` doesn't support custom results like this. You can return custom types like the code below, you can use exist response types like `ListResponse`, `RetrieveResponse`, etc. or you can extend these responses as well. Another way, you can create a standard API that doesn't extend from `ServiceEndpoint`.

```
[ScriptInclude]
public class TestClass
{
    //your object properties
}
```

You can use this class inside the endpoint as a response type.

**Reply by eissa1005** [reply]
OK , Thanks
